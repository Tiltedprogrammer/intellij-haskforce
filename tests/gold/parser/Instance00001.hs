module Instance00001 where

instance {-# OVERLAPPABLE #-} Foo a => Bar a where
  foo = bar
