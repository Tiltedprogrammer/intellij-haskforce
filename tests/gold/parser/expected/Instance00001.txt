Haskell File
  MODULEDECL
    PsiElement(HaskellTokenType.module)('module')
    QCONID
      CONID
        PsiElement(HaskellTokenType.conidRegexp)('Instance00001')
    PsiElement(HaskellTokenType.where)('where')
  BODY
    INSTANCEDECL
      PsiElement(HaskellTokenType.instance)('instance')
      PPRAGMA
        PsiElement(HaskellTokenType.{-#)('{-#')
        PsiElement(HaskellTokenType.PRAGMA)('OVERLAPPABLE')
        PsiElement(HaskellTokenType.#-})('#-}')
      CTYPE
        CLSCONTEXT
          ATYPE
            OQTYCON
              QTYCON
                TYCON
                  CONID
                    PsiElement(HaskellTokenType.conidRegexp)('Foo')
          ATYPE
            TYVAR
              VARID
                PsiElement(HaskellTokenType.varidRegexp)('a')
        PsiElement(HaskellTokenType.=>)('=>')
        CTYPE
          TYPEE
            ATYPE
              OQTYCON
                QTYCON
                  TYCON
                    CONID
                      PsiElement(HaskellTokenType.conidRegexp)('Bar')
            ATYPE
              TYVAR
                VARID
                  PsiElement(HaskellTokenType.varidRegexp)('a')
      PsiElement(HaskellTokenType.where)('where')
      IDECL
        FUNORPATDECL
          PAT
            VARID
              PsiElement(HaskellTokenType.varidRegexp)('foo')
          RHS
            PsiElement(HaskellTokenType.=)('=')
            EXP
              QVAR
                QVARID
                  VARID
                    PsiElement(HaskellTokenType.varidRegexp)('bar')